File: backend/app/schemas/scanner.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class ScannerFilter(BaseModel):
    min_relative_volume: Optional[float] = Field(None, ge=0)
    max_distance_to_ema20: Optional[float] = Field(None, ge=0)
    min_distance_to_ema50: Optional[float] = Field(None, ge=0)
    sector: Optional[str] = None
    industry: Optional[str] = None
    min_market_cap: Optional[float] = Field(None, ge=0)
    max_market_cap: Optional[float] = Field(None, ge=0)
    is_adr: Optional[bool] = None
    min_price: Optional[float] = Field(None, ge=0)
    max_price: Optional[float] = Field(None, ge=0)
    breakout: Optional[bool] = None


class ScanResult(BaseModel):
    id: int
    scan_name: str
    symbol: str
    scan_date: datetime
    metrics: dict

    class Config:
        from_attributes = True


class ScannerResponse(BaseModel):
    results: List[ScanResult]
    total: int

File: frontend/types/scanner.ts
export interface ScannerFilter {
  min_relative_volume?: number;
  max_distance_to_ema20?: number;
  min_distance_to_ema50?: number;
  sector?: string;
  industry?: string;
  min_market_cap?: number;
  max_market_cap?: number;
  is_adr?: boolean;
  min_price?: number;
  max_price?: number;
  breakout?: boolean;
}

export interface ScanResult {
  symbol: string;
  name: string;
  sector: string | null;
  price: number;
  relative_volume: number;
  distance_to_ema20: number;
  rsi: number;
}

export interface QuickScannerFilter {
  min_rvol?: number
  min_relative_volume?: number
  market_cap_range?: [number, number]
  price_range?: [number, number]
  sector?: string
  min_distance_high?: number
  min_distance_high_52w?: number
  max_distance_high_52w?: number
  min_distance_ema20?: number
  min_rs?: number
  min_volume?: number
  max_adr?: number
  gap_pct_range?: [number, number]
  consolidation_days?: number
  upcoming_earnings_days?: number
  has_earnings?: boolean
  max_bollinger_width?: number
  ema20_above_ema50?: boolean
  ema50_above_ema200?: boolean
}

export interface QuickFilter {
  id: string
  name: string
  description?: string
  filters: QuickScannerFilter
}
